src/mechanics/cell/Jacobian.h

```diff
@@ -21,53 +21,71 @@ class Jacobian
             switch (globalDimension)
             {
             case 1:
-                std::tie(mInvJacobian, mDetJacobian) = CalculateFixedSize<1>(nodeValues, derivativeShapeFunctions);
+            {
+                Eigen::Matrix<double, 1, 1> jacobian = CalculateFixedSize<1,1>(nodeValues, derivativeShapeFunctions);
+                mJacobian = jacobian;
+                mInvJacobian = jacobian.inverse();
+                mDetJacobian = jacobian.determinant();
                 break;
+            }
             case 2:
-                std::tie(mInvJacobian, mDetJacobian) = CalculateFixedSize<2>(nodeValues, derivativeShapeFunctions);
+            {
+                Eigen::Matrix<double, 2, 2> jacobian = CalculateFixedSize<2,2>(nodeValues, derivativeShapeFunctions);
+                mJacobian = jacobian;
+                mInvJacobian = jacobian.inverse();
+                mDetJacobian = jacobian.determinant();
                 break;
+            }
             case 3:
-                std::tie(mInvJacobian, mDetJacobian) = CalculateFixedSize<3>(nodeValues, derivativeShapeFunctions);
+            {
+                Eigen::Matrix<double, 3, 3> jacobian = CalculateFixedSize<3,3>(nodeValues, derivativeShapeFunctions);
+                mJacobian = jacobian;
+                mInvJacobian = jacobian.inverse();
+                mDetJacobian = jacobian.determinant();
                 break;
             }
+            }
         }
-        else
+        else if (globalDimension == interpolationDimension + 1)
         {
-            assert(globalDimension == interpolationDimension + 1);
             switch (globalDimension)
             {
             case 2:
             {
-                const int numRows = derivativeShapeFunctions.rows();
-                NodeValues nodeCopy = nodeValues;
-                // see CalculateFixedSize(...) for the Eigen::Map functionality
-                auto nodeBlockCoordinates =
-                        Eigen::Map<Eigen::Matrix<double, 2, Eigen::Dynamic>>(nodeCopy.data(), 2, numRows);
-                mDetJacobian = (nodeBlockCoordinates * derivativeShapeFunctions).norm();
+                Eigen::Matrix<double, 2, 1> jacobian = CalculateFixedSize<2,1>(nodeValues, derivativeShapeFunctions);
+                Eigen::Matrix<double, 2,2> extendedJacobian;
+                extendedJacobian.col(0) = jacobian;
+                Eigen::Vector2d normal( -jacobian(1,0), jacobian(0,0));
+                normal /= normal.norm();
```

Comment:
There is something like `Eigen::normalize()` that might be appropriate here. Plus, I like to initialize the values as const as possible, so I think the following will work
~~~
const Eigen::Vector2d normal = Eigen::Vector2d(-jac(1,0), jac(0,0)).normalize()
~~~
And then we can think about avoiding the variable `normal` completely and assign the expression directly to ` extendedJacobian.col(1)`. But in this case, it might increase readablilty.